pkg/collector/corechecks/snmp/snmp.go

```diff
@@ -66,14 +109,23 @@ func (c *Check) Configure(rawInstance integration.Data, rawInitConfig integratio
 	}
 	log.Debugf("SNMP configuration: %s", c.config.ToString())
 
-	c.singleDeviceCk, err = devicecheck.NewDeviceCheck(c.config, c.config.IPAddress)
-	if err != nil {
-		return fmt.Errorf("failed to create device check: %s", err)
+	if c.config.IsDiscovery() {
+		c.discovery = discovery.NewDiscovery(c.config)
+		c.discovery.Start()
+	} else {
+		c.singleDeviceCk, err = devicecheck.NewDeviceCheck(c.config, c.config.IPAddress)
+		if err != nil {
+			return fmt.Errorf("failed to create device check: %s", err)
+		}
 	}
-
 	return nil
 }
 
+// Cancel is called when check is unscheduled
+func (c *Check) Cancel() {
+	c.discovery.Stop()
```

Comment:
should this stop also running checks?

Reply:
You mean if `Check.Run()` is currently being run ?

`Check.Cancel()` here won't stop `Check.Run()` activity, but I think it's fine to let it finish normally like any other integration `Check.Run()`.